### PR TITLE
Win installer

### PR DESCRIPTION
**Pull request by tschneidereit**

### CODE REVIEW

**Comment by tschneidereit** [comment]
(I'll provide a proper description once this actually works.)

**Comment by kubkon** [comment]
Just out of curiosity, why do we need an installer on Windows? Is it because we want dependencies dynamically linked rather than statically? Or is there a different reason?

**Comment by tschneidereit** [comment]
> Just out of curiosity, why do we need an installer on Windows? Is it because we want dependencies dynamically linked rather than statically? Or is there a different reason?

These are still statically linked binaries, so no, that's not it.

On Windows there isn't on obvious place to drop binaries into, so you'd commonly create a directory under `Program Files` and, for applications that should be available on the command line, add that to `%PATH%`. That's not really something our users should have to do manually.

To be clear, we should definitely also have a Homebrew recipe, and eventually be in distro package managers and all that :)

**Comment by kubkon** [comment]
> > Just out of curiosity, why do we need an installer on Windows? Is it because we want dependencies dynamically linked rather than statically? Or is there a different reason?
> 
> These are still statically linked binaries, so no, that's not it.
> 
> On Windows there isn't on obvious place to drop binaries into, so you'd commonly create a directory under `Program Files` and, for applications that should be available on the command line, add that to `%PATH%`. That's not really something our users should have to do manually.
> 

Ah, that's fair enough!

> To be clear, we should definitely also have a Homebrew recipe, and eventually be in distro package managers and all that :)

Yep, that's a must! I'm sure @marmistrz and myself will be able to help out with `pacman` and homebrew respectively :-)

**Comment by tschneidereit** [comment]
> Yep, that's a must! I'm sure @marmistrz and myself will be able to help out with `pacman` and homebrew respectively :-)

That'd be fantastic!

File: .azure-pipelines.yml [hunk 1]
```diff
@@ -156,28 +156,30 @@ jobs:
...
       displayName: Copy binaries (*nix)
       condition: and(succeeded(), ne( variables['Agent.OS'], 'Windows_NT' ), eq( variables['build_type'], 'release' ))
-    - task: ArchiveFiles@2
-      inputs:
-        rootFolderOrFile: $(Build.BinariesDirectory)/bundle
-        includeRootFolder: false
-        archiveType: 'zip'
-        archiveFile: '$(Build.ArtifactStagingDirectory)/wasmtime-$(tagName)-x86_64-windows.zip'
-      displayName: Archive files (Win)
-      condition: and(succeeded(), eq( variables['Agent.OS'], 'Windows_NT' ), eq( variables['build_type'], 'release' ))
```
Comment (alexcrichton) [inline .azure-pipelines.yml]: I've found that for automated installation of tools it's sometimes nifty to still have a tarball-like installer for Windows that's not just the executable ones. Perhaps both an `*.msi` could be created as well as a `*.tar.xz` to match the other platforms.

(I'm assuming that some sort of installer website would smoth over "what to choose" here as well)

Comment (tschneidereit) [reply .azure-pipelines.yml]: Good point, yes. I left the zip file in place with the latest version here.

**Comment by tschneidereit** [comment]
@alexcrichton this is ready for review again. I added the `LICENSE` and `README.md` files to the installer and moved the executables into a `bin` folder, which seems to be what everyone's doing.

**Comment by tschneidereit** [comment]
(Oh, and I had a CI check running in my own repo and stupidly canceled this one. I just triggered [a new one](https://dev.azure.com/CraneStation/Wasmtime/_build/results?buildId=99))

**Pull request merged**